code/game/machinery/computer/card.dm

```diff
@@ -537,35 +642,75 @@ GLOBAL_VAR_INIT(time_last_changed_position, 0)
 
 				else if(t1 == "Unassigned")
 					inserted_modify_id.access -= get_all_accesses()
+					if(B)
+						for(var/each in inserted_modify_id.registered_account.payment_per_department)
+							if(SSeconomy.is_nonstation_account(each))
+								continue
+							B.active_departments &= ~SSeconomy.account_bitflags[each]
+							B.payment_per_department[each] = 0
+							B.bonus_per_department[each] = 0
+						B.active_departments &= ~SSeconomy.account_bitflags[ACCOUNT_COM_ID] // micromanagement
+						B.payment_per_department[ACCOUNT_CIV_ID] = PAYCHECK_MINIMAL // for the love of god, let them have minimal
+					if(R)
+						for(var/each in B.payment_per_department)
+							if(SSeconomy.is_nonstation_account(each))
+								continue
+							R.fields["active_dept"] &= ~SSeconomy.account_bitflags[each]
+						R.fields["active_dept"] &= ~DEPT_BITFLAG_COM  // micromanagement2
+
 					log_id("[key_name(usr)] unassigned and stripped all access from [inserted_modify_id] using [inserted_scan_id] at [AREACOORD(usr)].")
 
 				else
-					var/datum/job/jobdatum
-					for(var/jobtype in typesof(/datum/job))
-						var/datum/job/J = new jobtype
-						if(ckey(J.title) == ckey(t1))
-							jobdatum = J
-							updateUsrDialog()
-							break
-
+					var/datum/job/jobdatum = SSjob.GetJob(t1)
 					if(!jobdatum)
 						to_chat(usr, "<span class='error'>No log exists for this job.</span>")
+						stack_trace("bad job string '[t1]' is given through HoP console by '[ckey(usr)]'")
 						updateUsrDialog()
 						return
 
 					inserted_modify_id.access = ( istype(src, /obj/machinery/computer/card/centcom) ? get_centcom_access(t1) : jobdatum.get_access() )
+
+					// reseting theirs first
+					if(B)
+						for(var/each in inserted_modify_id.registered_account.payment_per_department)
+							if(SSeconomy.is_nonstation_account(each))
+								continue
+							B.active_departments &= ~SSeconomy.account_bitflags[each]
+							B.payment_per_department[each] = 0
+							B.bonus_per_department[each] = 0
+						B.active_departments &= ~SSeconomy.account_bitflags[ACCOUNT_COM_ID] // micromanagement
+					if(R)
+						for(var/each in B.payment_per_department)
+							if(SSeconomy.is_nonstation_account(each))
+								continue
+							R.fields["active_dept"] &= ~SSeconomy.account_bitflags[each]
+						R.fields["active_dept"] &= ~DEPT_BITFLAG_COM  // micromanagement2
```

Comment:
I dohn't know what all this stuff is, but I feel like it should really be procs on the data core or economy subsystem or bank account or something, seems really messy tbh

Reply:
I attached more descriptions instead, that explain what it's for. Making these as a proc makes it uglier.